src/EVM/SymExec.hs

```diff
@@ -834,19 +853,19 @@ equivalenceCheck' solvers branchesA branchesB create = do
     -- the solver if we can determine unsatisfiability from the cache already
     -- the last element of the returned tuple indicates whether the cache was
     -- used or not
-    check :: App m => UnsatCache -> Set Prop -> m EquivResult
-    check knownUnsat props = do
+    check :: App m => UnsatCache -> (Set Prop, String) -> m (EquivResult, String)
+    check knownUnsat (props, meaning) = do
```

Comment:
I am a bit confused here, why you need to pass the `meaning` parameter here if you don't do anything with it, you are just returning it with the original result.

Reply:
As always, you picked a great issue. I have fixed this now. This fix should make `checkAll` nicer too, I think. Thanks for this, it really was silly. Let me know what you think!